### PR TITLE
TF-351 Prevent showing of unstyled configuration page in Prestashop when clicking configure

### PR DESCRIPTION
**Pull request by aelchaninov**

### CODE REVIEW

File: tapfiliate/tapfiliate.php [hunk 1]
```diff
@@ -8,6 +8,8 @@ class Tapfiliate extends Module
 {
     const UPDATE_TYPE_UPDATE = 'order_update';
     const UPDATE_TYPE_NEW = 'order_new';
+    // @TODO: Production url
+    const TAPFILIATE_BASE_URL = 'https://app.dev.tap';
```
Comment (4overking) [inline tapfiliate/tapfiliate.php]: Use Prod host

File: tapfiliate/tapfiliate.php [hunk 2]
```diff
@@ -55,84 +58,108 @@ public function uninstall()
...
-            WebserviceKey::setPermissionForAccount($apiAccess->id, $permissions);
-
-            Configuration::set('TAP_WEBSERVICE_KEY_ID', $apiAccess->id);
+            try {
+                $client = new GuzzleHttp\Client();
+                // @TODO remove
+                $client->setDefaultOption('verify', false);
+                $response = $client
+                    ->post(
+                        self::TAPFILIATE_BASE_URL . '/integrations/prestashop/auth/check/',
```
Comment (4overking) [inline tapfiliate/tapfiliate.php]: Configuration::get('TAPFILIATE_BASE_URL') ?? self::TAPFILIATE_BASE_URL

File: tapfiliate/tapfiliate.php [hunk 3]
```diff
@@ -55,84 +58,108 @@ public function uninstall()
...
+    public function hookActionDispatcher($args)
+    {
+        if (
+            defined('_PS_ADMIN_DIR_')
+            && isset($args['controller_class'])
+            && $args['controller_class'] == 'AdminModulesController'
+            && Tools::getValue('configure') == 'tapfiliate'
+        ) {
+            $context = Context::getContext();
+            $shop = $context->shop;
```
Comment (4overking) [inline tapfiliate/tapfiliate.php]: useless variables, use class properties in the payload

File: tapfiliate/tapfiliate.php [hunk 4]
```diff
@@ -55,84 +58,108 @@ public function uninstall()
...
+                $response = $client
+                    ->post(
+                        self::TAPFILIATE_BASE_URL . '/integrations/prestashop/auth/check/',
+                        ['body' => $payload],
+                    )
+                    ->getBody()
+                    ->getContents();
+
+                Tools::redirect($response);
+            }  catch(\Exception $e) {
```
Comment (4overking) [inline tapfiliate/tapfiliate.php]: Catch only Guzzle`s errors

File: tapfiliate/tapfiliate.php [hunk 5]
```diff
@@ -55,84 +58,108 @@ public function uninstall()
...
+                'shop_id' => $shop_id,
+                'shop_group_id' => $shop_group_id,
+                'login_key' => Configuration::get('TAPFILIATE_LOGIN_KEY')
             ];
 
-            WebserviceKey::setPermissionForAccount($apiAccess->id, $permissions);
-
-            Configuration::set('TAP_WEBSERVICE_KEY_ID', $apiAccess->id);
+            try {
+                $client = new GuzzleHttp\Client();
```
Comment (4overking) [inline tapfiliate/tapfiliate.php]: move out the creation of the $client from the try block

File: tapfiliate/tapfiliate.php [hunk 6]
```diff
@@ -239,14 +266,14 @@ private function sendOrderUpdate($update_type, Order $order)
 
         try {
             $client = new GuzzleHttp\Client();
-            // @TODO: Production url
-            $client->post('https://a.dev.tap/integrations/prestashop/webhooks/receive/', [
+            // @TODO remove
+            $client->setDefaultOption('verify', false);
+            $client->post(self::TAPFILIATE_BASE_URL . '/integrations/prestashop/webhooks/receive/', [
```
Comment (4overking) [inline tapfiliate/tapfiliate.php]: Configuration::get('TAPFILIATE_BASE_URL') ?? self::TAPFILIATE_BASE_URL

File: tapfiliate/tapfiliate.php [hunk 7]
```diff
@@ -239,14 +266,14 @@ private function sendOrderUpdate($update_type, Order $order)
...
                 'body' => $payload,
                 'headers' => [
                     'X-Webhook-Signature' => $signature,
                     'X-Prestashop-Domain' => $shop->domain
-                ],
-                'verify' => false // @TODO
+                ]
             ]);
         } catch(\Exception $e) {
```
Comment (4overking) [inline tapfiliate/tapfiliate.php]: specify the exception

**Review by 4overking — CHANGES_REQUESTED** [review]

File: tapfiliate/tapfiliate.php (hunk 2 above)
Comment (aelchaninov) [reply tapfiliate/tapfiliate.php]: can you clarify please @4overking  `Configuration::get('TAPFILIATE_BASE_URL')`  wherefrom we would take this value? what is the reason to save such value in DB, not in plugin code?

File: tapfiliate/tapfiliate.php (hunk 1 above)
Comment (aelchaninov) [reply tapfiliate/tapfiliate.php]: later on it would be required, first we should test it on test environment with test URL at least...

**Review by 4overking — APPROVED** [review]

**Pull request merged**